contracts/FiatToken.cdc

```diff
@@ -1,1287 +1,268 @@
-                    let executor = self.ownerExecutorCapability!.borrow() ?? panic("cannot borrow OwnerExecutor capability")
-                    executor.reassignMasterMinter(to: to, newPath: path)
-                case "reassignBlocklister":
-                    let to = p.getArg(i: 0)! as? Address ?? panic("cannot downcast receiver address")
-                    let path = p.getArg(i: 1)! as? PrivatePath ?? panic("cannot downcast new link path")
-                    let executor = self.ownerExecutorCapability!.borrow() ?? panic("cannot borrow OwnerExecutor capability")
-                    executor.reassignBlocklister(to: to, newPath: path)
-                case "reassignPauser":
-                    let to = p.getArg(i: 0)! as? Address ?? panic("cannot downcast receiver address")
-                    let path = p.getArg(i: 1)! as? PrivatePath ?? panic("cannot downcast new link path")
-                    let executor = self.ownerExecutorCapability!.borrow() ?? panic("cannot borrow OwnerExecutor capability")
-                    executor.reassignPauser(to: to, newPath: path)
-                default:
-                    panic("Unknown transaction method")
-            }
-            destroy (p)
-            return nil
-        }
-
-        pub fun UUID(): UInt64 {
-            return self.uuid
-        }
-
-        pub fun getTxIndex(): UInt64 {
-            return self.multiSigManager.txIndex
-        }
-
-        pub fun getSignerKeys(): [String] {
-            return self.multiSigManager.getSignerKeys()
-        }
-        pub fun getSignerKeyAttr(publicKey: String): OnChainMultiSig.PubKeyAttr? {
-            return self.multiSigManager.getSignerKeyAttr(publicKey: publicKey)
-        }
-
-        destroy() {
-            destroy self.multiSigManager
-        }
-
-        init(pk: [String], pka: [OnChainMultiSig.PubKeyAttr]) {
-            self.multiSigManager <-  OnChainMultiSig.createMultiSigManager(publicKeys: pk, pubKeyAttrs: pka)
-            self.ownerExecutorCapability = nil
-        }
-    }
-
-    pub resource MasterMinterExecutor {
-
-        pub fun configureMinterController(minter: UInt64, minterController: UInt64) {
-            // Overwrite the minter if the MinterController is already configured (a MinterController can only control 1 minter)
-            FiatToken.managedMinters.insert(key: minterController, minter)
-            emit ControllerConfigured(controller: minterController, minter: minter)
-        }
-
-        pub fun removeMinterController(minterController: UInt64){
-            pre {
-                FiatToken.managedMinters.containsKey(minterController): "cannot remove unknown MinterController"
+                FiatToken.blocklist[self.uuid] == nil: "Vault Blocklisted"
             }
-            FiatToken.managedMinters.remove(key: minterController)
-            emit ControllerRemoved(controller: minterController)
+            self.balance = self.balance - amount
```

Comment:
it's nice to be working with fixed point numbers! Btw does this throw if amount > self.balance? Should there be a pre check?

Reply:
yes, there is a pre-condition in `FungibleToken.Vault.withdraw()` that checks the balance